packages/protocol/src/certificate/CertificateAuthority.ts

```diff
@@ -48,49 +65,64 @@ export class CertificateAuthority {
         return this.#construction;
     }
 
-    static async create(crypto: Crypto, options?: StorageContext | CertificateAuthority.Configuration) {
+    static async create(
+        crypto: Crypto,
+        options?: StorageContext | CertificateAuthority.Configuration | CertificateAuthority.PartialConfiguration,
+    ) {
         return asyncNew(CertificateAuthority, crypto, options);
     }
 
-    constructor(crypto: Crypto, options?: StorageContext | CertificateAuthority.Configuration) {
+    constructor(
+        crypto: Crypto,
+        options?: StorageContext | CertificateAuthority.Configuration | CertificateAuthority.PartialConfiguration,
+    ) {
         this.#crypto = crypto;
+
         this.#construction = Construction(this, async () => {
-            // Use provided CA config or read from storage, otherwise initialize and store
             const certValues = options instanceof StorageContext ? await options.values() : (options ?? {});
 
+            if (this.#isValidStoredRootCertificate(certValues)) {
+                const hasIcacInStorage = this.#isValidStoredIcacCertificate(certValues);
+                if (hasIcacInStorage) {
+                    this.#intermediateCert = true;
+                } else if (typeof certValues.intermediateCert === "boolean") {
+                    this.#intermediateCert = certValues.intermediateCert;
+                } else {
+                    this.#intermediateCert = false;
+                }
+
+                this.#loadFromStorage(certValues);
+
+                // Reset intermediateCert to false if set but no ICAC data exists
+                if (this.#intermediateCert && !hasIcacInStorage) {
+                    logger.warn(
+                        "intermediateCert=true in configuration but no valid ICAC data in storage. Resetting to 2-tier PKI.",
+                    );
+                    this.#intermediateCert = false;
+                }
```

Comment:
If an error is thrown above this can be removed IMO

Reply:
makes sense too